test_programs/compile_failure/impl_stricter_than_trait/src/main.nr

```diff
@@ -0,0 +1,32 @@
+trait Serialize<let N: u32> {
```

Comment:
I added this as I just wanted to see the output from the diagnostic. It is redundant to the `impl_stricter_than_trait ` test under `noirc_frontend/test.rs` though so I can remove it if we see fit.

Reply:
Might as well remove it, yeah. The frontend test is better anyway since we assert for the specific error case.

Reply:
I have removed this